UnityProject/Assets/Scripts/Systems/Antagonists/Antags/Changeling/ChangelingMain.cs

```diff
@@ -0,0 +1,643 @@
+
+		[Client]
+		private void SyncAbilityList(string oldValue, string newValue)
+		{
+			abilitesIDSNow = newValue;
+
+			if (AbilitiesNowDataSynced == null)
+				AbilitiesNowDataSynced = new List<ChangelingData>();
+			else
+				AbilitiesNowDataSynced.Clear();
+			foreach (string id in abilitesIDSNow.Split("\n"))
+			{
+				if (short.TryParse(id, out var idParsed) && !AbilitiesNowDataSynced.Contains(ChangelingAbilityList.Instance.FromIndex(idParsed)))
+					AbilitiesNowDataSynced.Add(ChangelingAbilityList.Instance.FromIndex(idParsed));
+			}
+		}
+
+		[Client]
+		private void SyncChemCount(float oldValue, float newValue)
+		{
+			chem = newValue;
+
+			try
+			{
+				if (PlayerManager.LocalPlayerScript != null && PlayerManager.LocalPlayerScript.Mind != null && changelingMindID == PlayerManager.LocalPlayerScript.Mind.netId)
+				{
+					if (changelingMind == null)
+						changelingMind = PlayerManager.LocalPlayerScript.Mind;
+					SetChangelingUI();
+					uiChangeling.UpdateChemText();
+				}
+			}
+			catch
+			{
+				Logger.LogError("Changeling can`t set up UI when sync chem", Category.Changeling);
+			}
+		}
+
+		[Client]
+		private void SyncMindID(uint oldValue, uint newValue)
+		{
+			changelingMindID = newValue;
+			try
+			{
+				if (PlayerManager.LocalPlayerScript != null && PlayerManager.LocalPlayerScript.Mind != null && changelingMindID == PlayerManager.LocalPlayerScript.Mind.netId)
+				{
+					if (changelingMind == null)
+						changelingMind = PlayerManager.LocalPlayerScript.Mind;
+					SetChangelingUI();
+				}
+			}
+			catch
+			{
+				Logger.LogError("Changeling can`t set up UI when sync mind", Category.Changeling);
+			}
+		}
+
+		[Client]
+		private void SyncEPCount(int oldValue, int newValue)
+		{
+			evolutionPoints = newValue;
+			try
+			{
+				if (PlayerManager.LocalPlayerScript != null && PlayerManager.LocalPlayerScript.Mind != null && changelingMindID == PlayerManager.LocalPlayerScript.Mind.netId)
+				{
+					if (changelingMind == null)
+						changelingMind = PlayerManager.LocalPlayerScript.Mind;
+					SetChangelingUI();
+					uiChangeling.UpdateEPText();
+				}
+			}
+			catch
+			{
+				Logger.LogError("Changeling can`t set up UI when sync ep", Category.Changeling);
+			}
+		}
+
+		[Client]
+		private void SyncResetCounts(int oldValue, int newValue)
+		{
+			resetCount = newValue;
+			try
+			{
+				UIManager.Display.hudChangeling.UpdateResetButton();
+			}
+			catch
+			{
+				Logger.LogError("Changeling failes to update reset button", Category.Changeling);
+			}
+		}
+
+		[Client]
+		private void SyncMaxResetCounts(int oldValue, int newValue)
+		{
+			resetCountMax = newValue;
+			try
+			{
+				UIManager.Display.hudChangeling.UpdateResetButton();
+			}
+			catch
+			{
+				Logger.LogError("Changeling failes to update reset button", Category.Changeling);
+			}
+		}
+
+		#endregion
+
+		#region Commands
+
+		/// <summary>
+		/// Resets changeling abilities
+		/// </summary>
+		[Command(requiresAuthority = false)]
```

Comment:
Giving a command with `requiresAuthority = false` is not recommended as other players can cheat to spam these commands either to cheat or ruin the experience for other players. 
If you're unable to rethink the structure/flow of how these functions are called, at least try putting in a few checks to make sure that this isn't as easily abused.